### PR TITLE
Add methods sum(), min() and max() to collections

### PR DESCRIPTION
**Pull request by matifreyre**
Implements methods requested on #691.

### CODE REVIEW

**Comment by flbulgarelli** [comment]
Looks good to me!

File: org.uqbar.project.wollok.tests/src/org/uqbar/project/wollok/tests/interpreter/CollectionTestCase.xtend [hunk 1]
```diff
@@ -296,4 +314,13 @@ class CollectionTestCase extends AbstractWollokInterpreterTestCase {
 		}'''.interpretPropagatingErrors
 	}
 	
+	@Test
+	def void sumNoArgs() {
+		'''
+		program p {
+			«instantiateCollectionAsNumbersVariable»
+			assert.equals(34, numbers.sum())
```
Comment (flbulgarelli) [inline org.uqbar.project.wollok.tests/src/org/uqbar/project/wollok/tests/interpreter/CollectionTestCase.xtend]: @matifreyre  Could we add tests for the edge case of empty and unitary collections?

Comment (matifreyre) [inline org.uqbar.project.wollok.tests/src/org/uqbar/project/wollok/tests/interpreter/CollectionTestCase.xtend]: We certainly can, but now that you mention it... min(closure) and
max(closure), which are the base for min() and max() respectively, use
absolute(closure, criteria) as their base. This method doesn't fail on
empty lists, but returns null. I'd rather have min and max (either version
of each one) fail, what do you think?

The case of sum() or sum(closure) is much simpler, of course.

Comment (flbulgarelli) [inline org.uqbar.project.wollok.tests/src/org/uqbar/project/wollok/tests/interpreter/CollectionTestCase.xtend]: Agree!

**Comment by matifreyre** [comment]
I think this one can be merged... any objections?

File: org.uqbar.project.wollok.tests/src/org/uqbar/project/wollok/tests/interpreter/CollectionTestCase.xtend [hunk 2]
```diff
@@ -53,6 +55,8 @@ class CollectionTestCase extends AbstractWollokInterpreterTestCase {
 		program p {
 			«instantiateCollectionAsNumbersVariable»		
 			assert.equals(22, numbers.max() )
+			assert.equals(1, [1].max())
+			assert.throwsException({[].max()})
 		}'''.interpretPropagatingErrors
```
Comment (flbulgarelli) [inline org.uqbar.project.wollok.tests/src/org/uqbar/project/wollok/tests/interpreter/CollectionTestCase.xtend]: As a suggestion for the future, it is better to add more tests for different scenarios than adding more assertion in the same test.
